File: FrontEnd/src/models/requestToHiredDTO.ts
import { UploadFile } from "antd/lib/upload/interface";
import { WorkflowInstancesDTO } from "@models/workflowInstancesDTO";
import { WorkflowStepDTO } from "@models/workflowStepDTO";

export interface RequestToHiredDTO {
  id: string;
  status: number | null;
  createdDate: Date;
  reason: string | null;
  quantity: number;
  filePath: string | null;
  idCategoryVacancies: string;
  Files: UploadFile[];
  workflowInstances: WorkflowInstancesDTO[];
  countWorkFlowStep: number;
  currentWorkFlowStep: WorkflowStepDTO;
  idUnit: string;
  createdBy: string;
}

File: BackEnd/ExamManagement/DPD.HR.Business.Dto/Dto/RequestToHiredDto.cs
using DPD.HumanResources.Entities.Entities;

namespace DPD.HumanResources.Dtos.Dto;

public class RequestToHiredDto
{
    public Guid Id { get; set; }
    public int? Status { get; set; }
    public DateTime? CreatedDate { get; set; }
    public string? Reason { get; set; }
    public int? Quantity { get; set; }
    public string? FilePath { get; set; }
    public Guid? IdCategoryVacancies { get; set; }
    public Guid CreatedBy { get; set; }
    public Guid IdUnit { get; set; }
    public Guid? IdFile { get; set; }

    public CategoryVacancies? CategoryVacancies { get; set; }
    public List<WorkflowInstances>? WorkflowInstances { get; set; }
    public int CountWorkFlowStep { get; set; }
    public WorkflowStep? CurrentWorkFlowStep { get; set; }

}